infra/blueprint-test/pkg/utils/asserthttp.go

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/go-testing-interface"
+)
+
+// AssertHTTP provides a collection of HTTP asserts.
+type AssertHTTP struct {
+	httpClient *http.Client
+}
+
+// NewAssertHTTP creates a new AssertHTTP.
+func NewAssertHTTP() *AssertHTTP {
+	return &AssertHTTP{http.DefaultClient}
+}
+
+// SetHTTPClient sets a substitute HTTP Client for assert requests.
+func (ah *AssertHTTP) SetHTTPClient(c *http.Client) *AssertHTTP {
+	if c != nil {
+		ah.httpClient = c
+	}
+	return ah
+}
+
+// AssertSuccessWithRetry runs httpRequest and retries on errors outside client control.
+func (ah *AssertHTTP) AssertSuccessWithRetry(t testing.T, r *http.Request) {
+	t.Helper()
+	Poll(t, ah.httpRequest(t, r), 3, 2*time.Second)
+}
+
+// AssertSuccess runs httpRequest without retry.
+func (ah *AssertHTTP) AssertSuccess(t testing.T, r *http.Request) error {
+	t.Helper()
+	_, err := ah.httpRequest(t, r)()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return err
+}
+
+// AssertResponseWithRetry runs httpResponse and retries on errors outside client control.
+func (ah *AssertHTTP) AssertResponseWithRetry(t testing.T, r *http.Request, wantCode int, want ...string) {
+	t.Helper()
+	Poll(t, ah.httpResponse(t, r, wantCode, want...), 3, 2*time.Second)
+}
+
+// AssertResponse runs httpResponse without retry.
+func (ah *AssertHTTP) AssertResponse(t testing.T, r *http.Request, wantCode int, want ...string) error {
+	t.Helper()
+	_, err := ah.httpResponse(t, r, wantCode, want...)()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return err
```

Comment:
If we are calling t.Fatal on err, is there a reason to return err too?

Reply:
Removed.